apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.ts

```diff
@@ -15,35 +15,50 @@ import { contains } from '@prairielearn/path-utils';
 import { getCourseOwners } from '../../lib/course.js';
 import * as editorUtil from '../../lib/editorUtil.js';
 import { FileDeleteEditor, FileRenameEditor, FileUploadEditor } from '../../lib/editors.js';
-import { getPaths } from '../../lib/instructorFiles.js';
+import { getPaths, type Paths } from '../../lib/instructorFiles.js';
 import { encodePath } from '../../lib/uri-util.js';
 
-import { InstructorFileBrowser } from './instructorFileBrowser.html.js';
+import {
+  type DirectoryListings,
+  type DirectoryEntryDirectory,
+  type DirectoryEntryFile,
+  type FileInfo,
+  InstructorFileBrowser,
+  InstructorFileBrowserNoPermission,
+} from './instructorFileBrowser.html.js';
 
 const router = Router();
 
 function isHidden(item: string) {
   return item[0] === '.';
 }
 
-async function browseDirectory(file_browser) {
-  const filenames = await fs.readdir(file_browser.paths.workingPath);
+async function browseDirectory({
+  paths,
+  has_course_permission_edit,
+  example_course,
+}: {
+  paths: Paths;
+  has_course_permission_edit: boolean;
+  example_course: boolean;
+}): Promise<DirectoryListings> {
+  const filenames = await fs.readdir(paths.workingPath);
   const ansiUp = new AnsiUp();
   const all_files = await async.mapLimit(
     filenames
       .sort()
       .map((name, index) => ({ name, index }))
       .filter((f) => !isHidden(f.name)),
     3,
-    async (file) => {
-      const filepath = path.join(file_browser.paths.workingPath, file.name);
+    async (file: { name: string; index: number }) => {
```

Comment:
Is `async.mapLimit` not correctly set up with generic types here?

Reply:
I tried a few options, and VSCode has been unable to identify the type of `file` here without a type annotation. Even using `async.mapLimit<...>(...` didn't work.

Reply:
Looks like this may be a TypeScript issue, I opened one here: https://github.com/microsoft/TypeScript/issues/59309